jetstream/kv.go

```diff
@@ -1122,11 +1208,20 @@ func (kv *kvs) WatchFiltered(ctx context.Context, keys []string, opts ...WatchOp
 
 		var op KeyValueOp
 		if len(m.Header) > 0 {
-			switch m.Header.Get(kvop) {
-			case kvdel:
-				op = KeyValueDelete
-			case kvpurge:
-				op = KeyValuePurge
+			if m.Header.Get(kvop) != "" {
+				switch m.Header.Get(kvop) {
+				case kvdel:
+					op = KeyValueDelete
+				case kvpurge:
+					op = KeyValuePurge
+				}
+			} else if m.Header.Get(MarkerReasonHeader) != "" {
```

Comment:
are we 100% sure we should treat those the same as headers?

Reply:
I was following ADR here: https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-48.md#retrieving-values

But I think that makes sense, this marks the same set of KV operations, only different mechanisms.